### DISCUSSION TITLE
PAGView可见性恢复后没有继续播放

### DISCUSSION THREAD

**Discussion by koxyoxyo** [Tencent/libpag#1897] [General] [answered]
版本4.2.100，当PAG不是无限循环播放时，PAGView隐藏后再恢复可见，PAGView没有继续播放PAG（以前版本是会自动恢复播放的）

断点看了一下，隐藏后会调用pause
<img width="792" alt="image" src="https://github.com/Tencent/libpag/assets/49026908/26c916d0-61ab-412d-b0b4-487ff689155d">
然后会走到notifyEnd中，并设置_isPlaying为false
<img width="881" alt="image" src="https://github.com/Tencent/libpag/assets/49026908/861111ac-8837-44e0-9943-3ccf45f8ae21">
<img width="609" alt="image" src="https://github.com/Tencent/libpag/assets/49026908/e10f4add-b4c9-44ee-be94-d9b029a0d5cc">

导致恢复可见时的resumeAnimator操作没有走到doplay
<img width="818" alt="image" src="https://github.com/Tencent/libpag/assets/49026908/ae1d69e3-0dcb-42cc-8e1d-f8396b46cf1b">

想请教一下这个问题是已知的吗，还是设计是如此？

**Comment by kevingpqi123** [comment]
感谢反馈，这个问题我们复现到了，如果不是无限循环，预期的效果是还未播放完进入后台再进入前台还是处于播放状态

**Reply by kevingpqi123** [reply, marked as answer]
可以升级到 4.3.33 版本，最新的版本有修复这个问题
